### PR TITLE
Add hashCommitment to public API

### PR DESCRIPTION
**Pull request by acolytec3**
Adds `hashCommitment` to public API to allow verkle node hashes to be generated

### CODE REVIEW

File: src.ts/tests/ffi.spec.ts [hunk 1]
```diff
@@ -218,4 +218,17 @@ describe('bindings', () => {
...
+    val[0] = 1
+    const oneCommit = verkleCrypto.updateCommitment(verkleCrypto.zeroCommitment, 0, new Uint8Array(32), val)
+    const oneHash = Uint8Array.from([
+      79, 26, 219, 175, 186, 122,   6, 119,
+     155, 65, 250, 228,  16, 187, 246,  40,
+     112, 99, 158, 244,  92,  49, 204, 106,
+      74, 47,  87, 231, 210,  64, 123,  10
+   ])
+   assert.deepEqual(verkleCrypto.hashCommitment(oneCommit), oneHash)
+  })
```
Comment (gabrocheleau) [inline src.ts/tests/ffi.spec.ts]: Could we add a comment that links to where this data comes from?

**Review by gabrocheleau — DISMISSED** [review]

**Comment by kevaundray** [comment]
This looks good to me; hashCommitment is the same method here in go-verkle: https://github.com/ethereum/go-verkle/blob/7d920df305f0f1fc135f3ed70443263c5fa62cf9/ipa.go#L64

Which just calls map_to_scalar_field under the hood and then serializes it.

We have unit tests for hashCommitment in that file and a seperate one for updateCommitment, I'm wondering if perhaps this new test is testing something different?

**Review by kevaundray — DISMISSED** [review]

**Review by acolytec3 — COMMENTED** [review]
>We have unit tests for hashCommitment in that file and a seperate one for updateCommitment, I'm wondering if perhaps this new test is testing something different? 

No, this is just me not looking hard enough in the tests for coverage of this method.  I'll remove it.

**Review by acolytec3 — COMMENTED** [review]
I have removed the duplicate test.  Also, I created an additional section for the `updateCommitment` test to reflect what I believe is the answer to my question in #50. @kevaundray, does this look correct?

**Comment by kevaundray** [comment]
> I have removed the duplicate test.  Also, I created an additional section for the `updateCommitment` test to reflect what I believe is the answer to my question in #50. @kevaundray, does this look correct?

Yep this looks perfect

**Review by gabrocheleau — APPROVED** [review]

**Comment by holgerd77** [comment]
Nit: I would be a fan if we have at least one functional test on the JS side for each function (even if it's a duplicate one), so that we can see that the API works end to end.

**Comment by holgerd77** [comment]
(and is e.g. properly exposed)

**Comment by acolytec3** [comment]
> Nit: I would be a fan if we have at least one functional test on the JS side for each function (even if it's a duplicate one), so that we can see that the API works end to end.

In this instance, the test already existed and I just didn't realize it.  For `hashCommitment`, [this](https://github.com/ethereumjs/verkle-cryptography-wasm/blob/89b405ec0a3de3599fe58285d2d9a9b55e91b00a/src.ts/tests/ffi.spec.ts#L104) test already existed and my work just duplicated it.

**Pull request merged**